specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json

```diff
@@ -226,6 +226,74 @@
         {
           "$ref": "#/definitions/IntegrationRuntime"
         }
+      ],
+      "properties": {
+        "typeProperties": {
+          "description": "When this property is not null, means this is a linked integration runtime. The property is used to access original integration runtime.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SelfHostedIntegrationRuntimeTypeProperties"
+        }
+      }
+    },
+    "SelfHostedIntegrationRuntimeTypeProperties": {
+      "description": "The self-hosted integration runtime properties.",
+      "type": "object",
+      "properties": {
+        "linkedInfo": {
```

Comment:
How about renaming linkedInfo to linkedIntegrationRuntimeAuthorization and the properties class to LinkedIntegrationRuntimeAuthorizationType, and the inherited classes to LinkedIntegrationRuntimeKeyAuthorization and  LinkedIntegrationRuntimeRbacAuthorization

Reply:
It is actually info here, not just authorization. we may add more properties here which are nothing related with authorization.

Reply:
Agreed to change properties class to LinkedIntegrationRuntimeType, inherited classes to LinkedIntegrationRuntimeKeyAuthorization and LinkedIntegrationRuntimeRbacAuthorization

Reply:
One more question here - Self Hosted IR can only have one linked IR? Shouldn't this be an array?

Reply:
linked IR doesn't exist in self-hosted IR as property, it has its own entity with selfhosted IR info